flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/tasks/AbstractInvokable.java

```diff
@@ -123,6 +126,17 @@ public boolean shouldInterruptOnCancel() {
 		return shouldInterruptOnCancel;
 	}
 
+	/**
+	 * If the invokable implementation executes user code in a thread other than,
+	 * {@link Task#getExecutingThread()}, this method returns the stack trace of that executing
+	 * thread.
+	 *
+	 * @see Task#getStackTraceOfExecutingThread()
+	 */
+	public Optional<StackTraceElement[]> getStackTraceOfExecutingThread() {
```

Comment:
nit: maybe change this to `Optional<Thread> getExecutingThread()`? It would feel somehow a bit cleaner for me

Reply:
Ok, `Optional<StackTraceElement[]>` feels awkward.

Reply:
Done.